### ISSUE TITLE
Incompatible indices error message seems to refer to incorrect version numbers

### ISSUE THREAD

**Issue by jdreyf** [pachterlab/kallisto#400]
Hi,

I got the following error message from kallisto quant, but kallisto seems to be on version 0.50.0 and it doesn't explain what those versions refer to. 
```
[quant] fragment length distribution will be estimated from the data
Error: incompatible indices. Found version 10, expected version 12
Rerun with index to regenerate"
```
I've also seen others get error message "Error: incompatible indices. Found version 0, expected version 10", e.g. [https://github.com/ENCODE-DCC/rna-seq-pipeline/issues/19](https://github.com/ENCODE-DCC/rna-seq-pipeline/issues/19), so I think there is an issue with the produced version numbers in this error message.

Thanks for the helpful software!
JD

**Comment by Yenaled** [comment]
kallisto version 0.50.0 utilizes an updated kallisto index. Therefore, if you made an index using an older version of kallisto, it will not work with kallisto 0.50.0.

As the message suggests, "rerun with index to regenerate" (means you have to run the "kallisto index" command on a reference transcriptome FASTA file to build a new index), as follows (example below uses 8 threads):

`kallisto index -i new_index.idx -t 8 reference_transcriptome.fasta`

**Comment by jdreyf** [comment]
Hi @Yenaled,

I agree that "Rerun with index to regenerate" is clear. But what I & others online find unclear is "Found version 10, expected version 12" or "Found version 559903, expected version 10". Presumably this sentence can be improved to avoid some of that confusion.

Thanks for all your work on kallisto,
JD

**Comment by Yenaled** [comment]
Yeah, don't worry about the version numbers -- it's just for our own internal use (and those numbers are arbitrary and simply to prevent memory corruption when someone uses a wrong index). Though if someone found version 559903, that probably means that their index file was majorly corrupted in some way haha or maybe they accidentally supplied something like a text file, pdf file, image file, etc. rather than an index.

Anyway, feel free to post any further questions about kallisto -- I'm usually super responsive on GitHub :)

**Comment by ijcbioinfo** [comment]
Hi,

I am still having the same issue with kallisto/0.50.1:

`kallisto quant --rf-stranded -i /mnt/beegfs/amerkel/MLLT3/tRNAseq/kallisto_index/hg38 -o CB-1-C -t 5 -b 30 fastq/CB-1-C.1.trim.paired.fastq fastq/CB-1-C.2.trim.paired.fastq`

```
[quant] fragment length distribution will be estimated from the data
Error: incompatible indices. Found version 0, expected version 13
Rerun with index to regenerate
```

I created the index with same version (I only have one version installed).

**Comment by ijcbioinfo** [comment]
Hi,

I am still having the same issue with kallisto/0.50.1:

```
kallisto quant --rf-stranded -i ~/tRNAseq/kallisto_index/hg38 -o CB-1-C -t 5 -b 30 fastq/CB-1-C.1.trim.paired.fastq fastq/CB-1-C.2.trim.paired.fastq

[quant] fragment length distribution will be estimated from the data
Error: incompatible indices. Found version 0, expected version 13
Rerun with index to regenerate

```
I created the index with same version (I only have one version installed).

**Comment by ijcbioinfo** [comment]
Sorry, I just figured it is matter of the index file not found. I gave the wrong file name. This works perfectly fine:

`kallisto quant --rf-stranded -i ~/tRNAseq/kallisto_index/hg38 -o CB-1-C -t 5 -b 30 fastq/CB-1-C.1.trim.paired.fastq fastq/CB-1-C.2.trim.paired.fastq`

May be you could improve the error message?
